docs/src/content/docs/basic-concepts.mdx

```diff
@@ -0,0 +1,116 @@
+title: Basic Concepts
+description: Understanding the core concepts of GenSX
+---
+
+GenSX is a workflow engine designed for building complex LLM applications. It's built around functional, reusable components that are composed using JSX to create and orchestrate complicated workflows.
+
+## Components
+
+Components are the building blocks of GenSX applications; they're pure TypeScript functions that:
+
+- Accept props as input
+- Return a single output or another component
+- Have zero side effects
+- Don't depend on external state
+- Are fully typed using TypeScript
+
+Here's an example of a simple component:
+
+```tsx
+interface GreetingProps {
+  name: string;
+}
+type GreetingOutput = string;
+
+const Greeting = gsx.Component<GreetingProps, GreetingOutput>(
+  async ({ name }) => {
+    return `Hello, ${name}!`;
+  },
+);
+```
+
+Components can also return other components using JSX:
+
+```tsx
+const AIGreeting = gsx.Component<GreetingProps, string>(async ({ name }) => {
+  return (
+    <ChatCompletion
+      model="gpt-4o-mini"
+      messages={[
+        {
+          role: "system",
+          content: "You are a friendly assistant that greets people warmly.",
+        },
+        { role: "user", content: `Write a greeting for ${name}.` },
+      ]}
+    />
+  );
+});
+```
+
+## JSX and Data Flow
+
+GenSX uses JSX to compose components into workflows. Data flows from parent to child through props allowing you to nest components and pass data between them. When components are combined, they will return the value of the innermost component.
+
+```tsx
+// Parent receives output from ChildComponent through the child function
+<ParentComponent input="some data">
+  {(parentResult) => <ChildComponent data={parentResult} />}
+</ParentComponent>
+```
+
+Unlike React's concurrent rendering model, GenSX components execute sequentially from top to bottom. Each component completes before its children begin.
+
+## Component Types
+
+GenSX provides two main types of components:
+
+1. **Regular Components** (`gsx.Component`) - For standard synchronous or asynchronous operations
+2. **Streaming Components** (`gsx.StreamingComponent`) - For handling streaming responses from LLMs
+
+For streaming components, you can enable streaming by passing the `stream` prop and consuming the result as an async iterator:
+
+```tsx
+const stream = await gsx.execute(
+  <MyStreamingComponent input="some data" stream={true} />,
+);
+
+// Process the streaming response
+for await (const chunk of stream) {
+  process.stdout.write(chunk);
+}
+```
+
+## Executing Components
+
+Components are executed using `gsx.execute()`, which processes the JSX tree from top to bottom:
+
+```tsx
+const result = await gsx.execute(<MyComponent input="some data" />);
+```
+
+Components can be tested and evaluated in isolation, making it easy to debug and verify individual steps of your workflow. This is particularly valuable when building complex LLM applications that need robust evaluation.
```

Comment:
Maybe we can make this a little bit more relatable. Being able to mock up inputs for a single component and test it in isolation is a wayyyyy faster dev loop that running an entire 20 step workflow over and over again that takes two minutes, not to mention cheaper as well!

This section is straightforward and compelling, but I think it undersells the impact a little bit.

Reply:
fair. rather than changing what's there, i'm thinking i'll add on a couple of sentences

--- 

Rather than having to run an entire workflow to test a change to a single component, you can test just that component, dramatically speeding up your dev loop. This isolation also makes unit testing more manageable, as you can create specific test cases without having to worry about the rest of the workflow.